app/main/views.py

```diff
@@ -16,11 +19,16 @@ def index():
     req_opt_desc = Descriptor.query.filter_by(
         id=req_opt_desc.descriptor_id
     ).first()
+    options = Descriptor.query.all()
+    options = [o for o in options if len(o.text_resources) == 0 and o.id is not req_opt_desc.descriptor_id]
```

Comment:
we should first check if `req_opt_desc` is not None. If it is we should default it to some -1 value and then check if `o.id != -1` in that case